src/toil/jobStores/googleJobStore.py

```diff
@@ -179,10 +188,28 @@ def assignID(self, jobDescription):
                   jobStoreID, '<no command>' if jobDescription.command is None else jobDescription.command)
         jobDescription.jobStoreID = jobStoreID
 
+    jobsPerBatchInsert = 200
+
+    @contextmanager
+    def batch(self):
+        self._batchedUpdates = []
+        yield
+
+        batches = [self._batchedUpdates[i:i + self.jobsPerBatchInsert] for i in
+                   range(0, len(self._batchedUpdates), self.jobsPerBatchInsert)]
+        for batch in batches:
+            for jobDescription in batch:
+                jobDescription.pre_update_hook()
+                self._writeBytes(jobDescription.jobStoreID, pickle.dumps(jobDescription, protocol=pickle.HIGHEST_PROTOCOL))
```

Comment:
I don't think this quite addresses the thrust of the issue.

In the AWS job store, when we batch up the updates, we collect all the updates that need to be made and then we make [one call into the AWS Python module](https://github.com/DataBiosphere/toil/blob/de3ecf6d22dea55752d1487ac9957527fc03aa27/src/toil/jobStores/aws/jobStore.py#L315) with the whole batch. The point of this is that the AWS module can more efficiently send the updates to the cloud when it has all of them at once, versus sending each one at a time.

Here, you've implemented the machinery to defer sending the updates to Google until you have them all collected, but then you're still looping over them and sending them to the Google Python module one at a time. Like you did with the deletion, you should be using `with self.storageClient.batch():` here around each batch.